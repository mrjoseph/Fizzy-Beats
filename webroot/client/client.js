import { ApolloLink } from 'apollo-link';
import { withClientState } from 'apollo-link-state';
import ApolloClient from 'apollo-client';
import { createUploadLink } from 'apollo-upload-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { persistCache } from 'apollo-cache-persist';
import { defaults } from './graphql/defaults/defaults';
import { resolvers } from './graphql/resolvers/resolvers';


const cache = new InMemoryCache();
// persistCache({
//   cache,
//   storage: window.localStorage,
// });


const stateLink = withClientState({
  defaults,
  resolvers, 
});

const uploadLink = createUploadLink({
  uri: 'http://localhost:5000/graphql',
});

const link = ApolloLink.from([stateLink, uploadLink]);


export const client = new ApolloClient({
  link,
  cache,
});

