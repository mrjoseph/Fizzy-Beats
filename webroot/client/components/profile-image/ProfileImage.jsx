import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import { ThemeProvider } from 'styled-components';
import { ProfileImageCropArea } from './profileImage.styles';

import { GET_USERS_QUERY } from '../../graphql/queries/queries';
class ProfileImage extends Component {
  render() {
    
    const { username, data, userId } = this.props;
    const theme = {
      width: '40px',
    };
    const { loading, error, profileId } = data;
    if(loading) return <div>loading</div>
    if(error) return <div>error</div>
    const profileImageURL =
    `http://localhost:3003/static/${userId}${profileId.profileImage}`;


    return (
      <ThemeProvider theme={theme}>
        <ProfileImageCropArea>
          <img src={profileImageURL} alt={username} />
        </ProfileImageCropArea>
      </ThemeProvider>
    );

  }
}
ProfileImage.propTypes = {
  userId: PropTypes.string.isRequired,
};

export default graphql(GET_USERS_QUERY, {
  name: 'data',
  options: (props) => {
    return {
      variables: {
        id: props.userId,
      },
    }
  },
})(ProfileImage);

