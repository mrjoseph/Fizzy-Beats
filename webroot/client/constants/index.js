const cdn = 'http://localhost:3003';

export default cdn;