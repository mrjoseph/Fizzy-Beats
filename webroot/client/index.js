import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from 'react-router-dom';
import { ApolloClient, createNetworkInterface } from 'react-apollo';

const networkInterface = createNetworkInterface({
  uri: 'http://localhost:5000/graphql'
});
const client = new ApolloClient({
  networkInterface: networkInterface
});

import Routes from './app';
const App = () => <BrowserRouter><Routes /></BrowserRouter>;
ReactDOM.render(
  <ApolloProvider client={client}>
    <App />
  </ApolloProvider>
  , document.getElementById("root"));