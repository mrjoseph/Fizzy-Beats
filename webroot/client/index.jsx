import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-client';
import { HttpLink } from 'apollo-link-http';
import { InMemoryCache } from 'apollo-cache-inmemory';

import Routes from './app';

const link = new HttpLink({
  uri: 'http://localhost:5000/graphql',
});
const client = new ApolloClient({
  link,
  cache: new InMemoryCache(),
});
const App = () => <Router><Routes /></Router>;
ReactDOM.render(
  <ApolloProvider client={client}>
    <App />
  </ApolloProvider>,
  document.getElementById('root'),
);
