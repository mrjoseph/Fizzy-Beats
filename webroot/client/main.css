body {
    font-family: Arial, Helvetica, sans-serif;
    font-size:100%;
    background-color: #f1f1f1;
}
