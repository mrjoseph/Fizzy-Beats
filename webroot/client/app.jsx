import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './views/home/home';
import User from './views/user/user';
import About from './views/about/about';
import NotFound from './views/not-found/not-found';
import './main.css';
const App = () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route exact path="/about" component={About} />
    <Route exact path="/user" component={User} />
    <Route component={NotFound} />
  </Switch>
);
export default App;
