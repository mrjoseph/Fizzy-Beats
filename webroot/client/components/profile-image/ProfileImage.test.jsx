import React from 'react';
import { createUploadLink } from 'apollo-upload-client';

import { ApolloClient } from 'apollo-client';
import { ApolloProvider } from 'react-apollo';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import { client } from '../../client';
import ProfileImage from './ProfileImage';
// global.fetch = require('jest-fetch-mock');
describe('ProfileImage', () => {
    it('should', () => {
        expect(1).toEqual(1)
    })
    describe('loading', () => {
        let createUploadLink;
        let component;
        const props = {
            username: 'Tony Stark',
            data: {
                loading: true,
                error: false, 
                profileId: null
            },
            userId: '123456789qwertyu'
        };
        beforeEach(() => {
            component = shallow(<ProfileImage {...props} />)
        });
        it('should render a loading div', () => {
            console.log(component);
        })
    })
    describe('success', () => {
        it('renders correctly', () => {
            const props = {
                username: 'Tony Stark',
                data: {
                    loading: false,
                    error: false, 
                    profileId: '123456789qwertyu'
                },
                userId: '123456789qwertyu'
            };
          const tree = renderer.create(
              <ApolloProvider client={client}>
                    <ProfileImage {...props} />
              </ApolloProvider>
          ).toJSON();
          expect(tree).toMatchSnapshot();
      });
      })
});