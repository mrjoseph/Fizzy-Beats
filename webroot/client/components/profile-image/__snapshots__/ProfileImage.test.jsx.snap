// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage success renders correctly 1`] = `
<div>
  loading
</div>
`;
